### PR TITLE
Add graphql query complexity histogram metric

### PR DESCRIPTION
**Pull request by AurelienFT**
## Linked Issues/PRs
Closes https://github.com/FuelLabs/fuel-core/issues/2274

## Checklist
- [x] Breaking changes are clearly marked as such in the PR description and changelog
- [x] New behavior is reflected in tests
- [x] [The specification](https://github.com/FuelLabs/fuel-specs/) matches the implemented behavior (link update PR if changes are needed)

### Before requesting review
- [x] I have reviewed the code myself
- [x] I have created follow-up issues caused by this PR and linked them here

### CODE REVIEW

**Review by xgreenx — APPROVED** [review]

File: crates/fuel-core/src/graphql_api/metrics_extension.rs [hunk 1]
```diff
@@ -118,4 +121,14 @@ impl Extension for MetricsExtInner {
 
         res
     }
+
+    async fn validation(
```
Comment (MitchTurner) [inline crates/fuel-core/src/graphql_api/metrics_extension.rs:125]: It doesn't look like this gets used by anything in this PR... where is it used?

**Review by MitchTurner — COMMENTED** [review]
One little question. Everything else looks good.

File: crates/fuel-core/src/graphql_api/metrics_extension.rs (hunk 1 above)
Comment (AurelienFT) [reply crates/fuel-core/src/graphql_api/metrics_extension.rs:125]: I'm not very familiar with async-graphql but the trait seems to have a default implementation and I override it. For me this is call by async-graphql itself when triggering validation step of the query

Comment (xgreenx) [reply crates/fuel-core/src/graphql_api/metrics_extension.rs:125]: It is how `async-graphql` library works. It has a bunch of default methods that are called during the processing of the request. Valuation is one of the phases there.
image

Comment (MitchTurner) [reply crates/fuel-core/src/graphql_api/metrics_extension.rs:125]: Ah. I missed it was a trait impl. That makes sense now.

**Review by MitchTurner — APPROVED** [review]

**Pull request merged**